core/server/master/src/main/java/alluxio/master/file/DefaultFileSystemMaster.java

```diff
@@ -2002,16 +2007,21 @@ private void renameInternal(RpcContext rpcContext, LockedInodePath srcInodePath,
 
     // Check options and determine if we should schedule async persist. This is helpful for compute
     // frameworks that use rename as a commit operation.
-    if (context.getPersist() && srcInode.isFile() && !srcInode.isPersisted()) {
-      mInodeTree.updateInode(rpcContext, UpdateInodeEntry.newBuilder()
-          .setId(srcInode.getId())
-          .setPersistenceState(PersistenceState.TO_BE_PERSISTED.name())
-          .build());
-      mPersistRequests.put(srcInode.getId(), new alluxio.time.ExponentialTimer(
-          ServerConfiguration.getMs(PropertyKey.MASTER_PERSISTENCE_INITIAL_INTERVAL_MS),
-          ServerConfiguration.getMs(PropertyKey.MASTER_PERSISTENCE_MAX_INTERVAL_MS),
-          srcInode.asFile().getPersistenceWaitTime(),
-          ServerConfiguration.getMs(PropertyKey.MASTER_PERSISTENCE_MAX_TOTAL_WAIT_TIME_MS)));
+    if (srcInode.isFile() && !srcInode.isPersisted()) {
+      long shouldPersistTime = srcInode.asFile().getShouldPersistTime();
+      if (context.getPersist() || shouldPersistTime == Constants.NO_AUTO_PERSIST_VALUE) {
```

Comment:
I think this condition should just be `context.getPersist`, its possible the user does not want to persist on rename.

Reply:
In Alluxio's integration with Hive and maybe other compute frameworks, will they be able to provide detail rename options like `context.getPersist` here?

Reply:
Yeah we can use the property key alluxio.user.file.persist.on.rename=true to get the desired behavior.

Reply:
Then I am good with only `context.getPersist`, thanks!